nixos/modules/services/networking/mbim-network.nix

```diff
@@ -0,0 +1,126 @@
+      postStart = ''
+        ${netCmd} status | grep -q 'Status: activated'
+        ${cliCmd} --query-ip-configuration
+      '';
+      postStop = "${netCmd} status | grep -q 'Status: deactivated'";
+    };
+in {
+  options.services.mbim-network = {
+    # Allow configuring zero or more MBIM devices with differing configurations.
+    devices = mkOption {
+      default = { };
+      example = literalExample ''"/dev/cdc-wdm0".apn = "internet";'';
+      description = ''
+        Each attribute of this option specifies an MBIM device (example:
+        "/dev/cdc-wdm0") which will be managed by a systemd unit (example:
+        "mbim-network-dev-cdc-wdm0".
+      '';
+      type = with types;
+        attrsOf (submodule {
+          options = {
+            apn = mkOption {
+              type = types.str;
+              example = "internet";
+              description = ''
+                The Access Point Name (APN) to use when starting mbim-network for
+                this device. For a list of common APN values, see
+                <link xlink:href="https://customer.cradlepoint.com/s/article/access-point-names-by-carrier"/>.
+              '';
+            };
+
+            apnUser = mkOption {
+              type = types.str;
+              default = "";
+              example = "user";
+              description =
+                "An optional username to use for APN authentication.";
+            };
+
+            apnPass = mkOption {
+              type = types.str;
+              default = "";
+              example = "password";
+              description = ''
+                An optional password to use for APN authentication. Note that any
+                password specified here will be persisted in the Nix store.
+              '';
+            };
+
+            apnAuth = mkOption {
+              type = types.enum [ "" "PAP" "CHAP" "MSCHAPV2" ];
+              default = "";
+              description = ''
+                An optional authentication protocol to use for APN authentication:
+                one of "" (none), "PAP", "CHAP", or "MSCHAPV2".
+              '';
+            };
+          };
+        });
+    };
+
+    proxy = mkOption {
+      type = types.bool;
+      # It seems that mbim-proxy greatly improves the responsiveness of
+      # mbimcli commands, so we should encourage users to use it by default.
+      default = true;
+      description =
+        "Whether or not to spawn mbim-proxy to proxy communications with MBIM devices.";
+    };
+  };
+
+  # For each configured device, create a systemd unit to run mbim-network.
+  config = mkIf (cfg.devices != { }) {
+    systemd.services = listToAttrs (mapAttrsFlatten
+      (name: value: nameValuePair (cleanName name) (makeSystemdUnit value name))
+      cfg.devices);
+
+    # TODO: consider adding systemd timers or similar to check the status of the
+    # connection on an ongoing basis and to restart this unit if need be.
```

Comment:
TODOs should be stuff that's really actionable. We don't know yet how stable it is, so could very well just drop this comment.